modifytags/gomodifytags.go

```diff
@@ -0,0 +1,830 @@
+package gomodifytags
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"io"
+	"io/ioutil"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/fatih/camelcase"
+	"github.com/fatih/structtag"
+	"golang.org/x/tools/go/buildutil"
+)
+
+// structType contains a structType node and it's name. It's a convenient
+// helper type, because *ast.StructType doesn't contain the name of the struct
+type structType struct {
+	name string
+	node *ast.StructType
+}
+
+// output is used usually by editors
+type output struct {
+	Start  int      `json:"start"`
+	End    int      `json:"end"`
+	Lines  []string `json:"lines"`
+	Errors []string `json:"errors,omitempty"`
+}
+
+// Config defines how tags should be modified
```

Comment:
(but it actually defines more than that -- see comments elsewhere).

As mentioned below, I think we should expose a much smaller API. But whatever we DO expose will need more comprehensive documentation.